product_main_supplier/models/product_template.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from datetime import date
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    # fixme could product.main_seller_id be used instead? it seems that
+    # “seller” and “supplier” are used interchangeably in this
+    # class. is this on purpose?
+
+    # "top" is different from "main":
+    # top_supplierinfo_id is a technical field,
+    # and it's not the main supplier
+    # that is selected but the first one.
+    # So top_supplier reflect more the reality.
```

Comment:
@victor-champonnois I did wrong thing about this. You can rename `top_supplierinfo` to `main_supplierinfo`. See #459